apps/prairielearn/src/pages/partials/questionAwardedPoints.ejs

```diff
@@ -1,11 +1,28 @@
 <% include('pointsFormatter'); %>
 
-<% for (var i = 0; i < awarded_points.length; i++) { %>
-    <% if (awarded_points[i] < init_points) { %>
-        <% if (awarded_points[i] > 0) { %>
-            <span class="badge badge-warning"><%= getStringFromFloat(awarded_points[i]) %></span>
+<% const MAX_DISPLAYED_VARIANTS = 10; %>
+
+<% if (previous_variants) { %>
+    <%
+        const collapseId = `variants-points-collapse-${instance_question_id}`;
+        const collapseButtonId = `variants-points-collapse-button-${instance_question_id}`;
+        let startHiddenIndex = null;
+        let endHiddenIndex = null;
+    %>
+    <% if (previous_variants.length > MAX_DISPLAYED_VARIANTS) { %>
+        <a id="<%= collapseButtonId %>" role="button" onclick="$(`#<%= collapseId %>`).css('display','inline'); $(`#<%= collapseButtonId %>`).css('display','none');">&hellip;</a>
+        <% startHiddenIndex = 0; %>
+        <% endHiddenIndex = previous_variants.length - MAX_DISPLAYED_VARIANTS; %>
+    <% } %>
+    <% for (var i = 0; i < previous_variants.length; i++) { %>
+        <% if (i == startHiddenIndex) { %>
+            <span id="<%= collapseId %>" style="display:none">
+        <% } %>
+        <% if (i == endHiddenIndex) { %>
+            </span>
         <% } %>
-    <% } else { %>
-        <span class="badge badge-success"><%= getStringFromFloat(awarded_points[i]) %></span>
+        <span class="badge badge-secondary">
+            <a class="text-white" href="<%= urlPrefix %>/instance_question/<%= instance_question_id %>/?variant_id=<%= previous_variants[i].variant_id %>"><%= Math.floor(previous_variants[i].max_submission_score * 100) + "%"  %></a> 
+        </span>
```

Comment:
I believe the `a` should be outside the `span`, otherwise only the text of the badge is a link, not the badge itself.

Reply:
Good point, it should just be a single `<a class="badge">`. Fixed in f4793b734a97d9fe1002466a3c3e770101f697d3